src/WebJobs.Script.WebHost/FileMonitoringService.cs

```diff
@@ -369,5 +382,41 @@ internal static async Task SetAppOfflineState(string rootPath, bool offline)
                 }, maxRetries: 3, retryInterval: TimeSpan.FromSeconds(1));
             }
         }
+
+        private class SuspendRestartRequest : IDisposable
+        {
+            private FileMonitoringService _fileMonitoringService;
+
+            public SuspendRestartRequest(FileMonitoringService fileMonitoringService)
+            {
+                _fileMonitoringService = fileMonitoringService;
+                if (Interlocked.Increment(ref _fileMonitoringService._suspensionRequestsCount) == 1)
+                {
+                    _fileMonitoringService._typedLogger.LogDebug("Restart is suspended");
```

Comment:
Can we update the log message to something like `"Entering restart suspension scope. ({0} requests)."`, showing the number of suspension requests/subscribers

Reply:
We can move this logic to the FileMonitoringService, so this type just calls methods on it.